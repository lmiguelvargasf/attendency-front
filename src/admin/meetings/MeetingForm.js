import React from 'react'
import { Form, TimePicker, Button, DatePicker, Input, Select } from 'antd'

const { Option } = Select

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 8 }
}
const tailLayout = {
  wrapperCol: { offset: 8, span: 8 }
}

const MeetingForm = ({ meeting, onFinish, onFinishFailed, projects }) => {
  return (
    <Form
      data-testid='create-meeting-form'
      {...layout}
      name='meeting'
      initialValues={meeting}
      onFinish={onFinish}
      onFinishFailed={onFinishFailed}
    >
      {meeting ? (
        <Form.Item
          label='Project'
          name='project'
          data-testid='disabled-input'
        >
          <Input disabled />
        </Form.Item>
      ) : (
        <Form.Item
          label='Project'
          name='project'
          rules={[{ required: true, message: 'Please select project!' }]}
          data-testid='select-project'
        >
          <Select allowClear>
            {projects.map(project => <Option key={project.key} value={project.key}>{project.title}</Option>)}
          </Select>
        </Form.Item>
      )}
      <Form.Item
        label='Date'
        name='date'
        rules={[{ required: true, message: "Please input meeting's date!" }]}
      >
        <DatePicker />
      </Form.Item>
      <Form.Item
        label='Time'
        name='time'
        rules={[{ required: true, message: "Please input meeting's time!" }]}
      >
        <TimePicker format='HH:mm' />
      </Form.Item>

      <Form.Item {...tailLayout}>
        <Button type='primary' htmlType='submit' data-testid='create-save-button'>
          {meeting ? 'Save' : 'Create'}
        </Button>
      </Form.Item>
    </Form>
  )
}

export default MeetingForm
