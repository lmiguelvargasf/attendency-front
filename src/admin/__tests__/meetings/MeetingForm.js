import React from 'react'
import { render, cleanup } from '@testing-library/react'
import MeetingForm from '../../meetings/MeetingForm'
import moment from 'moment'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

describe('MeetingForm component', () => {
  afterEach(cleanup)

  describe('meeting is passed', () => {
    let component
    beforeEach(() => {
      const meeting = {
        project: 'My Project',
        date: moment('2020-04-30', 'YYYY-MM-DD'),
        time: moment('17:00', 'HH:mm')
      }
      component = (
        <MeetingForm meeting={meeting} onFinish={jest.fn()} onFinishFailed={jest.fn()} />
      )
    })
    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays button saying Save and disabled input and not select', () => {
      const { getByTestId, queryByTestId } = render(component)
      expect(getByTestId('create-save-button')).toHaveTextContent(/^Save$/)
      expect(getByTestId('disabled-input')).not.toBeNull()
      expect(queryByTestId('select-project')).toBeNull()
    })
  })

  describe('meeting is not passed', () => {
    let component

    beforeEach(() => {
      const projects = [
        { key: 1, title: 'Project 1' },
        { key: 2, title: 'Project 2' }
      ]
      component = (
        <MeetingForm onFinish={jest.fn()} onFinishFailed={jest.fn()} projects={projects} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays button saying Create and select and not disabled input', () => {
      const { getByTestId, queryByTestId } = render(component)
      expect(getByTestId('create-save-button')).toHaveTextContent(/^Create$/)
      expect(getByTestId('select-project')).not.toBeNull()
      expect(queryByTestId('disabled-input')).toBeNull()
    })
  })
})
