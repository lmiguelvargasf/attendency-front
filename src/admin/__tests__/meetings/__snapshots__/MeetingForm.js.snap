// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingForm component meeting is not passed matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    data-testid="create-meeting-form"
    id="meeting"
  >
    <div
      class="ant-row ant-form-item"
      data-testid="select-project"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="meeting_project"
          title="Project"
        >
          Project
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="meeting_project_list_0"
                    aria-autocomplete="list"
                    aria-controls="meeting_project_list"
                    aria-haspopup="listbox"
                    aria-owns="meeting_project_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="meeting_project"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                />
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="meeting_date"
          title="Date"
        >
          Date
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  placeholder="Select date"
                  readonly=""
                  size="12"
                  title=""
                  value=""
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="calendar"
                    class="anticon anticon-calendar"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="calendar"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="meeting_time"
          title="Time"
        >
          Time
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  placeholder="Select time"
                  readonly=""
                  size="10"
                  title=""
                  value=""
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="clock-circle"
                    class="anticon anticon-clock-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="clock-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              data-testid="create-save-button"
              type="submit"
            >
              <span>
                Create
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`MeetingForm component meeting is passed matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    data-testid="create-meeting-form"
    id="meeting"
  >
    <div
      class="ant-row ant-form-item"
      data-testid="disabled-input"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="meeting_project"
          title="Project"
        >
          Project
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input ant-input-disabled"
              disabled=""
              id="meeting_project"
              type="text"
              value="My Project"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="meeting_date"
          title="Date"
        >
          Date
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  placeholder="Select date"
                  readonly=""
                  size="12"
                  title="2020-04-30"
                  value="2020-04-30"
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="calendar"
                    class="anticon anticon-calendar"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="calendar"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-picker-clear"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="meeting_time"
          title="Time"
        >
          Time
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  placeholder="Select time"
                  readonly=""
                  size="10"
                  title="17:00"
                  value="17:00"
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="clock-circle"
                    class="anticon anticon-clock-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="clock-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-picker-clear"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              data-testid="create-save-button"
              type="submit"
            >
              <span>
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
